src/ccontrol/UserQueryFactory.cc

```diff
@@ -52,6 +52,7 @@
 #include "ccontrol/UserQueryType.h"
 #include "css/CssAccess.h"
 #include "css/KvInterfaceImplMem.h"
+#include "czar/Czar.h"
```

Comment:
You seem to need this to get access to the chunk map and to the interface class for the REgistry as per:
```
   auto cz = czar::Czar::getCzar();
    auto czarChunkMap = cz->getCzarChunkMap();
    auto czarRegistry = cz->getCzarRegistry();
```

I'm not sure this is right from the dependency standpoint as `Czar` is the front-end for handling incoming queries. `Czar` is also responsible for creating a branch of resources and services needed for processing queries. These resources are hosted by  `UserQuerySharedResources`. Hence, I would argue that you could add both pointers (chunk map and the registry API) to that collection to make it available to the downstream classes.

Reply:
I'll delete these lines for now as they aren't being used until the next ticket.

You bring up a good point. Part of why I did made the CzarRegistry class is that it could be destoryed and recreated to stop and restart the communications with the registry if there were problems. I don't know if it useful at this time, but restarting that communications seems like a good option to have.

Reply:
> Having lots of copies of that shared pointer could make destroying the existing object tricky...

Not at all. The standard pattern here is to introduce the "shutdown" (or "stop") method:
```
void CzarRegistry::shutdown() {
    _loop = false;
}
```
In this case, you could repurpose your flag for telling threads when it's time to equity. 
This would also help with the unit testing of the service.

A good example is the `qhttp` server:
```
class Server : public std::enable_shared_from_this<Server> {
public:
    using Ptr = std::shared_ptr<Server>;

    static Ptr create(...);

    ...

    //----- start() opens the server listening socket and installs the head of the asynchronous event
    //      handler chain onto the asio::io_service provided when the Server instance was constructed.
    //      Server execution may be halted either calling stop(), or by calling asio::io_service::stop()
    //      on the associated asio::io_service.

    void start();

    //----- stop() shuts down the server by closing all active sockets, including the server listening
    //      socket.  No new connections will be accepted, and handlers in progress will err out the next
    //      time they try to read/write from/to their client sockets.  A call to start() will be needed to
    //      resume server operation.

    void stop();
```
This interface works really well for unit testing.